org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/quickfix/DiagnosticResolution.java

```diff
@@ -34,12 +41,22 @@ public class DiagnosticResolution {
 
 	private IModification<EObject> modification;
 
+	private final ITextModification<Diagnostic, EObject, Document, List<TextEdit>> textModification;
+
 	private Factory factory;
 
 	private int relevance;
 
 	private Options options;
 
+	public DiagnosticResolution(final String label, final Factory modificationContextFactory, final ITextModification<Diagnostic, EObject, Document, List<TextEdit>> textModification) {
```

Comment:
No need for final in method parameters

Reply:
Would also need an `@since` I suppose

Reply:
You probably want a second constructor such that the `relevance` can be set

Reply:
done.